src/main/java/io/codelex/flightplanner/FlightServiceDB.java

```diff
@@ -0,0 +1,70 @@
+package io.codelex.flightplanner;
+
+import io.codelex.flightplanner.domain.Airport;
+import io.codelex.flightplanner.domain.Flight;
+import io.codelex.flightplanner.dto.FlightRequest;
+import io.codelex.flightplanner.dto.PageResult;
+import io.codelex.flightplanner.dto.SearchFlightsRequest;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+
+@Service
+@ConditionalOnProperty(prefix = "flightsapp", name = "appmode", havingValue = "database")
+public class FlightServiceDB implements FlightService {
```

Comment:
Ok skatos, ka šeit nav pabeigts, tad taisīšu review, kad būs pabeigts.
Uzreiz ietiekums, ko redzēju kā kļūdu daudzos citos darbos, nevajag no DB izgūt visus ierakstus, lai meklētu piemēram Flight, jāmeklē ir DB pusē (jebšu citiem vārdiem sakot repository.findAll() drīkst izmantot tikai tad, kad jāatgriež visi Flight un nekur citur):
Izlasi šito un arī zemāk tajā pašā rakstā ir manual custom queries:
https://www.baeldung.com/the-persistence-layer-with-spring-data-jpa#1-automatic-custom-queries
Kā ari ir šāds vairants:
https://www.baeldung.com/spring-data-query-by-example#examples